cranelift-wasm/src/state.rs

```diff
@@ -179,6 +179,11 @@ pub struct TranslationState {
     // `FuncEnvironment::make_direct_func()`.
     // Stores both the function reference and the number of WebAssembly arguments
     functions: HashMap<FuncIndex, (ir::FuncRef, usize)>,
+
+    // Track values that are introduced with SIMD's v128.const instruction; because cranelift has no
+    // v128 type, we type these values as I8x16 and use this mapping to raw_bitcast to the correct
+    // type once the value is used
+    constants: HashMap<Value, bool>,
```

Comment:
Does the same problem apply to `load`? Would it work to just unconditionally bitcast any SIMD value that doesn't have the desired SIMD type?

Reply:
Yeah, in my work-ahead branch I basically started doing just that--when functions return a `v128` then the original type of any SIMD instruction is lost and must be re-cast to the expected type when it is next used. I will remove this commit and add the code I have in my other branch.